connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -143,15 +156,12 @@ public static <T> HttpResponse<T> httpRequest(String url, String method, HttpHea
         } catch (IOException | InterruptedException | TimeoutException | ExecutionException e) {
             log.error("IO error forwarding REST request: ", e);
             throw new ConnectRestException(Response.Status.INTERNAL_SERVER_ERROR, "IO Error trying to forward REST request: " + e.getMessage(), e);
+        } catch (ConnectRestException e) {
```

Comment:
I can see why we're catching and re-throwing this kind of exception here, and I'm also having trouble thinking of anything significantly cleaner, so this might be good enough as-is. But do you think a brief comment might help other developers understand the intent here? The control flow is a little tricky.

Reply:
sure, added a comment, let me know if the wording is understandable